src/kernel/coinstats.cpp

```diff
@@ -93,7 +93,7 @@ static void ApplyHash(T& hash_obj, const Txid& hash, const std::map<uint32_t, Co
 {
     for (auto it = outputs.begin(); it != outputs.end(); ++it) {
         COutPoint outpoint = COutPoint(hash, it->first);
-        Coin coin = it->second;
```

Comment:
given the above warning, can you create a godbolt reproducer to check if the compiler can already deduce this or not?